### ISSUE TITLE
Yardmap does not rotate with building when the building is rotated for construction with the [ ] keys

### ISSUE THREAD

**Issue by Beherith** [beyond-all-reason/spring#91]
So always the default rotation of the yardmap is used (not the correctly rotated one) when checking for buildability.

**Comment by Beherith** [comment]
A good example is armadvsol, buildable by armck.
![image](https://user-images.githubusercontent.com/109391/137124078-c082d92b-98f2-492a-909c-6f01e9f624f5.png)
It has a square footprint, but a yardmap of 
yardmap = "oooo oooo yooy yooy",

The yardmap does not rotate when you rotate the building placement with [ ]

Very important to note: There are no buildings that have non-square yardmaps with "y" flags in them. So I tested armadvsol with:

		footprintx = 4,
		footprintz = 5,
		yardmap = "oooo oooo oooo yooy yooy",

This resulted in the most peculiar thing happenning on rotate:
When the building was facing up or down , the yardmap was not rotated, but at least semi-correct:
![image](https://user-images.githubusercontent.com/109391/137124900-21bc32b9-889a-403a-967d-39dd42e13aa5.png)

However, when the yardmap was rotated to face west or east, it seemed like the visual just remapped the yardmap to a transposed footprint:

![image](https://user-images.githubusercontent.com/109391/137125040-62e6662a-4b5b-4ee2-8940-147a8924a67d.png)


And additional corollary to all this, is that the "h" flag at the start of a yardmap, seems to work, and me specifying a pretty stupid testing high-res 4x4 yardmap: 

yardmap = "h yoyoyoyo yoyoyoyo yoyoyoyo yoyoyoyo oyoyoyoy oyoyoyoy oyoyoyoy oyoyoyoy",
"   h - 'High Resolution' - Indicates that the yardmap will use double resolution with four times as many characters. Must be the first character, and does not count as a square itself (New in version 93.0),"

This should allow adv solars to be offset below one another, like so:
![image](https://user-images.githubusercontent.com/109391/137126229-380a7d50-d067-414f-b526-61dadd54c2cf.png)

TLDR:  Once the building is constructed, the blocking status of the terrain is correct for the yardmap, but while placing it, is the yardmap incorrectly rotated.

**Comment by lhog** [comment]
Cool, rotation was indeed not taken into account. I had hoped this would be transformed by the engine beforehand. Not the case apparently.
Thanks for details, will fix.

**Comment by lhog** [comment]
`yardmap = "oooo oooo ooyy ooyy",` as a testing sample

South:
![screen02069](https://user-images.githubusercontent.com/1476261/137168899-5c7caf83-8851-4e5b-b885-615b2c22ff95.png)

North:
![screen02070](https://user-images.githubusercontent.com/1476261/137168907-621e5ca3-35d3-4928-882f-c46eee3e28b8.png)

East:
![screen02071](https://user-images.githubusercontent.com/1476261/137168912-aac6cd0e-d094-487a-986e-eacfe5b31962.png)

West:
![screen02072](https://user-images.githubusercontent.com/1476261/137168917-aa5e024f-4ac8-45ad-b21e-c75257cb3d6e.png)
